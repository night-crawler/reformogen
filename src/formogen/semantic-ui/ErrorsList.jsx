import React from 'react';
import PropTypes from 'prop-types';
import { isEmpty } from 'lodash';
import { Message } from 'semantic-ui-react';


ErrorsList.displayName = 'ErrorsList';
ErrorsList.propTypes = {
  header: PropTypes.string,
  messages: PropTypes.arrayOf(PropTypes.string),
  color: PropTypes.string,
};
ErrorsList.defaultProps = {
  header: 'Errors',
  color: 'red',
};
export function ErrorsList(props) {
  if (isEmpty(props.messages))
    return null;

  return (
    <Message color={ props.color }>
      { props.header && <Message.Header>{ props.header }</Message.Header> }
      <Message.List>
        { props.messages.map((msg, i) => 
          <Message.Item key={ i }>{ msg }</Message.Item>) }
      </Message.List>
    </Message>
  );
}
