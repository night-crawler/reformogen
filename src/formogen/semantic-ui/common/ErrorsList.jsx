import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Message } from 'semantic-ui-react';


ErrorsList.displayName = 'ErrorsList';
ErrorsList.propTypes = {
  header: PropTypes.string,
  messages: PropTypes.arrayOf(PropTypes.string),
  color: PropTypes.string,
};
ErrorsList.defaultProps = {
  header: 'Errors',
  color: 'red',
};
export function ErrorsList(props) {
  if (_.isEmpty(props.messages))
    return null;

  return (
    <Message color={ props.color }>
      { props.header && <Message.Header>{ props.header }</Message.Header> }
      <Message.List>
        { props.messages.map((msg, i) => 
          <Message.Item key={ i }>{ msg }</Message.Item>) }
      </Message.List>
    </Message>
  );
}
