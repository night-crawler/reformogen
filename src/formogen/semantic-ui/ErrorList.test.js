import { ErrorsList } from './ErrorsList';


it('<ErrorsList />', () => {
  const wrapper = shallow(
    <ErrorsList
      header={ 'test' }
      messages={ ['test1', 'test2'] }
      color='green'
    />
  );
  expect(wrapper).toMatchSnapshot();
});
