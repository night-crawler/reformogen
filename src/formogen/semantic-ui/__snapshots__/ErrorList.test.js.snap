// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorsList /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorsList
    color="green"
    header="test"
    messages={
      Array [
        "test1",
        "test2",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <MessageHeader>
          test
        </MessageHeader>,
        <MessageList
          as="ul"
        >
          <MessageItem
            as="li"
          >
            test1
          </MessageItem>
          <MessageItem
            as="li"
          >
            test2
          </MessageItem>
        </MessageList>,
      ],
      "color": "green",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "ul",
          "children": Array [
            <MessageItem
              as="li"
            >
              test1
            </MessageItem>,
            <MessageItem
              as="li"
            >
              test2
            </MessageItem>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "as": "li",
              "children": "test1",
            },
            "ref": null,
            "rendered": "test1",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "as": "li",
              "children": "test2",
            },
            "ref": null,
            "rendered": "test2",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MessageHeader>
            test
          </MessageHeader>,
          <MessageList
            as="ul"
          >
            <MessageItem
              as="li"
            >
              test1
            </MessageItem>
            <MessageItem
              as="li"
            >
              test2
            </MessageItem>
          </MessageList>,
        ],
        "color": "green",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "ul",
            "children": Array [
              <MessageItem
                as="li"
              >
                test1
              </MessageItem>,
              <MessageItem
                as="li"
              >
                test2
              </MessageItem>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "as": "li",
                "children": "test1",
              },
              "ref": null,
              "rendered": "test1",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "as": "li",
                "children": "test2",
              },
              "ref": null,
              "rendered": "test2",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
